### PR TITLE
Release 0.2 preparation

### PR DESCRIPTION
**Pull request by josh146**

### CODE REVIEW

File: mlir/llvm-project
Comment (josh146) [inline mlir/llvm-project:1]: @erick-xanadu should these changes to the submodules be showing here? 🤔

File: setup.py [hunk 1]
```diff
@@ -28,8 +28,7 @@
     jax_version = [line[4:].strip() for line in f.readlines() if "jax=" in line][0]
 
 requirements = [
-    # TODO: Change to "pennylane>=0.31" when next version releases.
-    "pennylane @ git+https://github.com/PennyLaneAI/pennylane/@master",
+    "pennylane>=0.31",
```
Comment (josh146) [inline setup.py:31]: Are there any other places we will need to make changes like this?

File: frontend/catalyst/_version.py [hunk 2]
```diff
@@ -16,4 +16,4 @@
    Version number (major.minor.patch[-label])
 """
 
-__version__ = "0.2.0-dev"
+__version__ = "0.2.0"
```
Comment (josh146) [inline frontend/catalyst/_version.py:19]: are there any other version strings where we will need to make version bumps?

File: mlir/llvm-project
Comment (erick-xanadu) [reply mlir/llvm-project:1]: I'm not sure how you merged the recent changes, but this is normal when updating the LLVM and MHLO submodules. See for example [here](https://github.com/PennyLaneAI/catalyst/commit/43e09f7d8ee66567bb5bdd9a2e261b6d85da1310#diff-0b11058d8332fcc13f6d23c90d7e3c92d7b1f7011b7ef4cf2e8fc34361110923). Did you just use `git merge`? If so, it should be fine.

Comment (josh146) [reply mlir/llvm-project:1]: yep! I just merged `main` into this branch.

Comment (erick-xanadu) [reply mlir/llvm-project:1]: It is fine. I'll resolve this now.

File: doc/changelog.md [hunk 3]
```diff
@@ -139,95 +91,262 @@
...
-      qml.RY(x, wires=0)
-      A = np.array(
-          [[complex(1.0, 0.0), complex(2.0, 0.0)], [complex(2.0, 0.0), complex(1.0, 0.0)]]
-      )
-      return qml.expval(qml.Hermitian(A, wires=0))
-
-  circuit(1.0)
+  Internally, the quantum instructions are generating OpenQASM3 kernels at
+  runtime; these are then executed on both local (``braket.local.qubit``) and
+  remote (``braket.aws.qubit``) devices backed by Amazon Braket Python SDK,
```
Comment (maliasadi) [inline doc/changelog.md:99]:
```suggestion
  runtime; these are then executed on both local (`braket.local.qubit`) and
  remote (`braket.aws.qubit`) devices backed by Amazon Braket Python SDK,

```

File: doc/changelog.md [hunk 4]
```diff
@@ -139,95 +91,262 @@
...
+  ```
+
+  ```pycon
+  >>> x = jnp.array([0.54, 0.31])
+  >>> cost(x)
+  array([0.80842369, 0.19157631])
+  ```
+
+* Support for returning the variance of Hamiltonians,
+  Hermitian matrices, and Tensors via ``qml.var``
```
Comment (maliasadi) [inline doc/changelog.md:225]:
```suggestion
  Hermitian matrices, and Tensors via `qml.var`

```

File: doc/changelog.md [hunk 5]
```diff
@@ -139,95 +91,262 @@
...
+  In the future, this will allow the ability to generate MLIR or LLVM-IR by
+  loading input from a string or file, rather than generating it from Python.
 
-* Handle C++ exceptions without unwinding the whole stack.
-  [#99](https://github.com/PennyLaneAI/catalyst/pull/99)
+  As part of this refactor, the following changes were made:
 
-* Support constant negative step sizes in ``@for_loop`` loops.
-  [#129](https://github.com/PennyLaneAI/catalyst/pull/129)
+  - Passes are now classes. This allow developers/users looking to change
```
Comment (maliasadi) [inline doc/changelog.md:313]:
```suggestion
  - Passes are now classes. This allows developers/users looking to change

```

File: doc/changelog.md [hunk 6]
```diff
@@ -236,31 +355,33 @@
...
+* Fix a memory regression and reduce memory footprint by removing unnecessary
+  temporary buffers. [#100](https://github.com/PennyLaneAI/catalyst/pull/100)
 
-* Provide a new abstraction to the ``QuantumDevice`` interface in the runtime called ``MemRefView``.
-  C++ implementations of the interface can iterate through and directly store results into the
-  ``MemRefView`` independant of the underlying memory layout. This can eliminate redundant buffer
-  copies at the interface boundaries, which has been applied to existing devices.
-  [#109](https://github.com/PennyLaneAI/catalyst/pull/109)
+* Provide a new abstraction to the `QuantumDevice` interface in the runtime
+  called `MemRefView`. C++ implementations of the interface can iterate
```
Comment (maliasadi) [inline doc/changelog.md:362]:
```suggestion
  called `DataView`. C++ implementations of the interface can iterate

```

File: doc/changelog.md [hunk 7]
```diff
@@ -236,31 +355,33 @@
...
+  through and directly store results into the `MemRefView` independant of the
```
Comment (maliasadi) [inline doc/changelog.md:363]:
```suggestion
  through and directly store results into the `DataView` independant of the

```

File: doc/changelog.md [hunk 8]
```diff
@@ -236,31 +355,33 @@
...
-  [#112](https://github.com/PennyLaneAI/catalyst/pull/112)
+* Reduce memory utilization by transferring ownership of buffers from the
+  runtime to Python instead of copying them. This includes adding a compiler
+  pass that copies global buffers into the heap as global buffers cannot be
+  transferred to Python. [#112](https://github.com/PennyLaneAI/catalyst/pull/112)
 
 * Temporary fix of use-after-free and dependency of uninitialized memory.
   [#121](https://github.com/PennyLaneAI/catalyst/pull/121)
 
 * Fixes file renaming within pass pipelines.
```
Comment (maliasadi) [inline doc/changelog.md:375]:
```suggestion
* Fix file renaming within pass pipelines.

```

File: doc/changelog.md [hunk 9]
```diff
@@ -236,31 +355,33 @@
...
   [#126](https://github.com/PennyLaneAI/catalyst/pull/126)
 
-* Fixes the issue with the ``do_queue`` deprecation warnings in PennyLane.
+* Fixes the issue with the `do_queue` deprecation warnings in PennyLane.
```
Comment (maliasadi) [inline doc/changelog.md:378]:
```suggestion
* Fix the issue with the `do_queue` deprecation warnings in PennyLane.

```

File: doc/changelog.md [hunk 10]
```diff
@@ -236,31 +355,33 @@
...
   [#146](https://github.com/PennyLaneAI/catalyst/pull/146)
 
-* Fixes the issue with gradients failing to work with ``jnp.array`` as constants.
-  [#152](https://github.com/PennyLaneAI/catalyst/pull/152)
+* Fixes the issue with gradients failing to work with hybrid functions that
```
Comment (maliasadi) [inline doc/changelog.md:381]:
```suggestion
* Fix the issue with gradients failing to work with hybrid functions that

```

File: frontend/catalyst/_version.py (hunk 2 above)
Comment (maliasadi) [reply frontend/catalyst/_version.py:19]: No. There are in fact two places that we use version strings: 1) `_version.py` 2) `changelog.md`

File: setup.py (hunk 1 above)
Comment (maliasadi) [reply setup.py:31]: No, that should be it.

**Review by maliasadi — COMMENTED** [review]
Thanks @josh146! :star_struck:

File: doc/changelog.md [hunk 11]
```diff
@@ -139,95 +91,262 @@
...
+  >>> x = jnp.array([0.1, 0.2])
+  >>> tangent = jnp.array([0.3, 0.6])
+  >>> jvp(x, tangent)
+  [array([0.09983342, 0.04      , 0.02      ]),
+   array([0.29850125, 0.24000006, 0.12      ])]
+  ```
+
+* Support for multiple backend devices within a single qjit-compiled function
+  is now available.
+  [#89](https://github.com/PennyLaneAI/catalyst/pull/89)
```
Comment (maliasadi) [inline doc/changelog.md:180]:
```suggestion
  [#86](https://github.com/PennyLaneAI/catalyst/pull/86)
  [#89](https://github.com/PennyLaneAI/catalyst/pull/89)

```

**Review by erick-xanadu — APPROVED** [review]
Looks great! Thanks Josh!

File: doc/changelog.md (hunk 4 above)
Comment (josh146) [reply doc/changelog.md:225]: nice catches :)
